tests/unit/test_multipan_rcp.cpp

```diff
@@ -0,0 +1,795 @@
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <openthread/config.h>
+
+#include <memory>
+
+#include "common/array.hpp"
+#include "common/code_utils.hpp"
+#include "common/instance.hpp"
+
+#include "ncp/ncp_base.hpp"
+#include "openthread/link_raw.h"
+
+#include "test_platform.h"
+#include "test_util.hpp"
+
+using namespace ot;
+using namespace ot::Ncp;
+
+enum
+{
+    kTestBufferSize = 800
+};
+
+enum
+{
+    kTestMacScanChannelMask = 0x01
+};
+
+OT_TOOL_PACKED_BEGIN
+struct RadioMessage
+{
+    uint8_t mChannel;
+    uint8_t mPsdu[OT_RADIO_FRAME_MAX_SIZE];
+} OT_TOOL_PACKED_END;
+
+static struct RadioMessage sDefaultMessage;
+static otRadioFrame        sDefaultFrame;
+
+static otRadioFrame *sTxFrame;
+
+otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance) { return sTxFrame; }
+
+class TestNcp : public NcpBase
+{
+public:
+    explicit TestNcp(ot::Instance *aInstance)
+        : mLastHeader(0)
+        , mLastStatus(0)
+        , NcpBase(aInstance)
+    {
+        memset(mMsgBuffer, 0, kTestBufferSize);
+        mTxFrameBuffer.SetFrameAddedCallback(HandleFrameAddedToNcpBuffer, this);
+        mTxFrameBuffer.SetFrameRemovedCallback(nullptr, this);
+    };
+
+    static void HandleFrameAddedToNcpBuffer(void                    *aContext,
+                                            Spinel::Buffer::FrameTag aTag,
+                                            Spinel::Buffer::Priority aPriority,
+                                            Spinel::Buffer          *aBuffer)
+    {
+        OT_UNUSED_VARIABLE(aTag);
+        OT_UNUSED_VARIABLE(aPriority);
+
+        static_cast<TestNcp *>(aContext)->HandleFrameAddedToNcpBuffer(aBuffer);
+    }
+
+    void HandleFrameAddedToNcpBuffer(Spinel::Buffer *aBuffer)
+    {
+        static const size_t display_size = 64;
+
+        memset(mMsgBuffer, 0, kTestBufferSize);
+        SuccessOrQuit(aBuffer->OutFrameBegin());
+        aBuffer->OutFrameRead(kTestBufferSize, mMsgBuffer);
+        SuccessOrQuit(aBuffer->OutFrameRemove());
+
+        // DumpBuffer("Received Buffer", mMsgBuffer, display_size);
+
+        updateSpinelStatus();
+    }
+
+    void Receive(uint8_t *aBuffer, size_t bufferSize) { HandleReceive(aBuffer, static_cast<uint16_t>(bufferSize)); }
+
+    void processTransmit()
+    {
+        LinkRawTransmitDone(sTxFrame, nullptr, OT_ERROR_NONE);
+        // printf("Transmit Finished\n");
+        processPendingCommands(); // Pending commands tasklet is posted by Transmit Done callback but not handled
+    };
+
+    void processEnergyScan()
+    {
+        LinkRawEnergyScanDone(Radio::kInvalidRssi);
+        // printf("Energy Scan Finished\n");
+        // processPendingCommands(); Pending commands are handled in Energy Scan Done callback
```

Comment:
Do we need this commented code?

Reply:
I don't think that we do. I will either remove these lines, or convert them into proper comments.